providers/openshift/provider.go

```diff
@@ -586,3 +580,18 @@ func request(client *http.Client, req *http.Request) (*simplejson.Json, error) {
 	}
 	return data, nil
 }
+
+func getKubeAPIURL() *url.URL {
+	host := "https://" + os.Getenv("KUBERNETES_SERVICE_HOST")
```

Comment:
Can this env actually change during a single running instance of the proxy?

Reply:
There is no support in the kubelet to send updated env vars.  I am not sure if it even possible to change the env for a running process.

Reply:
So that is my question then, what is the difference between fetching KUBERNETES_SERVICE_HOST at start-up vs. each time the URL is needed?

Reply:
Ah right well known should only be fetched once.

Reply:
We're only reading the ENV var here, are you proposing to have it cached?